File: Gun2/2.React/2.eTicaret/eTicaretServer/Models/Product.cs
namespace eTicaretServer.Models;

public sealed class Product
{
    public Product()
    {
        Id = Guid.NewGuid();
    }
    public Guid Id { get; set; }
    public string Name { get; set; } = default!;
    public string ImageUrl { get; set; } = default!;
    public int Stock { get; set; }
    public decimal Price { get; set; }
}

File: Gun5/e-ticaret-next-app/models/product.js
import mongoose from "mongoose";

const ProductSchema = new mongoose.Schema({
  id: { type: String},
  name: {
    type: String,
    required: true,
  },
  imageUrl: {
    type: String,
    required: true,
  },
  price: {
    type: Number,
    required: true
  },
  stock: {
    type: Number,
    required: true,
  },
});

export default mongoose.models.Product || mongoose.model("Product", ProductSchema);